csrc/device_lower/utils.cpp

```diff
@@ -762,6 +763,13 @@ bool isScalarExpr(Expr* expr) {
   return true;
 }
 
+bool isExtentEqualToMaxParallelTypeExtent(const IterDomain* id) {
+  auto& parallel_dim_map = GpuLower::current()->parallelDimensionMap();
```

Comment:
@zasdfgbnm do you think it makes sense to add an additional pre-condition, one where we check if domain is thread dim (`id->isThreadDim()`)?